tasks/fs-default.yml

```diff
@@ -6,19 +6,28 @@
   package:
     name: xfsprogs
     state: present
-  when: volume.fs_type == "xfs" and not ansible_check_mode
+  register: xfsprogs_installed
+  when: "volume.fs_type == 'xfs' and ['xfsprogs'] is not subset(ansible_facts.packages.keys()) \
+         and xfsprogs_installed is undefined and not ansible_check_mode"
```

Comment:
Might want to add a check for `volume.state == 'present'` since we only need ext2 and xfs tools to create/resize -- not to remove.

Reply:
Updated. About a quarter of the time, the `setup` module was being skipped after installing `lvm2` when absent in the user's system. I moved the `lvm_installed` variable from a registered variable from the `package` command to a fact so the `setup` module will always run.